src/legacy/core_plugins/kibana/public/dashboard/dashboard_app_controller.tsx

```diff
@@ -405,6 +410,75 @@ export class DashboardAppController {
       $scope.appState.$newFilters = [];
     };
 
+    $scope.onQuerySaved = savedQuery => {
+      $scope.savedQuery = savedQuery;
+    };
+
+    $scope.onSavedQueryUpdated = savedQuery => {
+      $scope.savedQuery = savedQuery;
+    };
+
+    $scope.onClearSavedQuery = () => {
+      delete $scope.savedQuery;
+      dashboardStateManager.setSavedQueryId(undefined);
+      queryFilter.setFilters([]);
```

Comment:
nit: You should use `queryFilter.removeAll()`

Reply:
👍 763f350